test/tap/update-index.js

```diff
@@ -1,200 +1,209 @@
+    t.ifError(err, 'registry mocked successfully')
     npm.load({ cache: CACHE_DIR, registry: common.registry }, function (err) {
-      if (extra) {
-        Object.keys(extra).forEach(function (k) {
-          npm.config.set(k, extra[k], 'user')
-        })
-      }
-      t.ifError(err, 'no error')
+      t.ifError(err, 'npm loaded successfully')
+      npm.config.set('cache', CACHE_DIR)
+      cacheBase = cacheFile(npm.config.get('cache'))(common.registry + '/-/all')
+      cachePath = path.join(cacheBase, '.cache.json')
       server = s
-      t.end()
+      t.pass('all set up')
+      t.done()
     })
   })
-}
-
-function cleanup (t) {
-  server.close(function () {
-    rimraf.sync(PKG_DIR)
-
-    t.end()
-  })
-}
-
-test('setup basic', function (t) {
-  setup(t, mocks.basic)
 })
 
-test('request basic', function (t) {
-  updateIndex(0, null, null, noFilter, function (er) {
-    t.ifError(er, 'no error')
-    t.end()
+test('updateIndex full request', function (t) {
+  setup()
+  var updated = Date.now()
+  server.get('/-/all').once().reply(200, {
+    '_updated': 1234,
+    'bar': { name: 'bar', version: '1.0.0' },
+    'foo': { name: 'foo', version: '1.0.0' }
+  }, {
+    date: updated
+  })
+  updateIndex(600, function (err, stream) {
+    if (err) throw err
+    t.ok(stream, 'returned a stream')
+    var results = []
+    stream.on('data', function (pkg) {
+      results.push(pkg)
+    })
+    ms.finished(stream, function (err) {
+      if (err) throw err
+      t.deepEquals(results, [{
+        name: 'bar',
+        version: '1.0.0'
+      }, {
+        name: 'foo',
+        version: '1.0.0'
+      }])
+      var fileData = JSON.parse(fs.readFileSync(cachePath))
+      t.ok(fileData, 'cache contents written to the right file')
+      t.deepEquals(fileData, {
+        '_updated': 1234,
+        bar: {
+          name: 'bar',
+          version: '1.0.0'
+        },
+        foo: {
+          name: 'foo',
+          version: '1.0.0'
+        }
+      }, 'cache contents based on what was written')
+      server.done()
+      cleanup()
+      t.end()
+    })
   })
 })
 
-test('cleanup basic', cleanup)
-
-test('setup auth', function (t) {
-  setup(t, mocks.auth)
-})
-
-test('request auth failure', function (t) {
-  updateIndex(0, null, null, noFilter, function (er) {
-    t.ok(er, 'got an error')
-    t.equals(er.code, 'E401', 'gotta get that auth')
-    t.ok(/^unauthorized/.test(er.message), 'unauthorized message')
-    t.end()
+test('updateIndex cache only', function (t) {
+  setup()
+  var now = Date.now()
+  var cacheContents = {
+    '_updated': now,
+    bar: { name: 'bar', version: '1.0.0' },
+    cool: { name: 'cool', version: '1.0.0' },
+    foo: { name: 'foo', version: '2.0.0' },
+    other: { name: 'other', version: '1.0.0' }
+  }
+  var fixture = new Tacks(File(cacheContents))
+  fixture.create(cachePath)
+  // TODO - figure out how to get the time stuff in here
```

Comment:
This comment predates a lot of stuff I ended up doing to make doubleplus sure these timestamps were good in the tests. I've removed it.